components/register-cluster/register_cluster.sh

```
@@ -0,0 +1,31 @@
+set -euo pipefail
+
+# Get cluster ingress domain
+INGRESS_DOMAIN=$(oc -n openshift-ingress-operator get ingresscontrollers.operator.openshift.io default -o jsonpath='{.status.domain}' 2>/dev/null || echo "apps.example.com")
+
+# Create Argo manager cluster SA
+kubectl create serviceaccount cluster-argocd-manager -n kube-system --dry-run=client -o yaml | kubectl apply -f -
+kubectl create clusterrolebinding cluster-argocd-manager-binding --clusterrole=admin --serviceaccount=kube-system:cluster-argocd-manager --dry-run=client -o yaml | kubectl apply -f -
+kubectl apply -f components/register-cluster/cluster_sa_token.yaml
+
+# Set Server variables
+export SERVER_URL=$(kubectl config view --minify -o jsonpath='{.clusters[0].cluster.server}' 2>/dev/null)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)